src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11692,7 +11766,8 @@ void Assembler::vex_prefix(Address adr, int nds_enc, int xreg_enc, VexSimdPrefix
   }
 }
 
-int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc, InstructionAttr *attributes) {
+int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc, InstructionAttr *attributes, bool src_is_gpr) {
```

Comment:
We could add an assert here:
if (src_is_gpr && src_enc >= 16) {
  assert(UseAPX, "APX features not enabled");
}

Reply:
Thanks, done.